File: rating-service/src/main/java/com/example/rating_service/model/dto/RatingRequestDto.java
package com.example.rating_service.model.dto;

import jakarta.validation.constraints.DecimalMax;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Digits;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;

import java.math.BigDecimal;
import java.util.UUID;

@Builder
public record RatingRequestDto(
        @NotNull(message = "ID оценивающего не может быть пустым")
        UUID raterUserId,

        @NotNull(message = "ID цели не может быть пустым")
        UUID targetUserId,

        @NotNull(message = "ID поездки обязателен")
        UUID rideId,

        @NotNull(message = "Рейтинг обязателен")
        @Digits(integer = 1, fraction = 2, message = "Рейтинг должен иметь максимум 1 знак до запятой и 2 знака после")
        @DecimalMin(value = "0.0", message = "Минимальный рейтинг — 0.0")
        @DecimalMax(value = "5.0", message = "Максимальный рейтинг — 5.0")
        Integer rating
) {
}

File: rating-service/src/main/kotlin/com/example/rating_service/dto/RatingRequestDto.kt
package com.example.rating_service.dto

import com.example.rating_service.enums.UserType
import jakarta.validation.constraints.DecimalMax
import jakarta.validation.constraints.DecimalMin
import jakarta.validation.constraints.NotNull
import java.util.UUID


data class RatingRequestDto(
    @field:NotNull(message = "field ride id is required")
    val rideId: UUID,

    @field:NotNull(message = "field rating is required")
    @field:DecimalMin(value = "0.0", inclusive = false, message = "rating must be > 0")
    @field:DecimalMax(value = "5.0", message = "rating must be ≤ 5")
    val rating: Int,

    @field:NotNull(message = "field target user id is required")
    val targetUserId: UUID,

    @field:NotNull(message = "field target user type is required")
    val targetUserType: UserType
)